File: app/src/main/java/com/demo/justapp/exchanger/data/network/RestApi.java
package com.demo.justapp.exchanger.data.network;

import com.demo.justapp.exchanger.models.remote.RatesResponse;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Интерфейс API для работы с сервером
 *
 * @author Sergey Rodionov
 */
public interface RestApi {

    /**
     * @param defaultCurrency валюта по умолчанию
     * @return возвращает список курсов валют
     */
    @GET("latest")
    Single<RatesResponse> getRates(@Query("base") String defaultCurrency);
}

File: app/src/main/kotlin/com/demo/justapp/exchanger/data/network/RestApi.kt
package com.demo.justapp.exchanger.data.network

import com.demo.justapp.exchanger.models.remote.RatesResponse
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Интерфейс API для работы с сервером
 *
 * @author Sergey Rodionov
 */
interface RestApi {

    @GET("latest")
    fun getRates(@Query("base") defaultCurrency: String): Single<RatesResponse>
}